src/kubeobjects/query.go

```diff
@@ -0,0 +1,24 @@
+package kubeobjects
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type kubeQuery struct {
+	kubeClient client.Client
+	kubeReader client.Reader
```

Comment:
I didn't find any usage of this member other than passing around.

Reply:
Its a leftover, the old `CreateOrUpdateSecret...` used to read the secret, for that it used the `kubeReader`, but the new logic tries to create it and if fails then it will just update it (which I think is simpler and therefore nicer)

but this `kubeQuery` should be reused for other `kubeobject`stuff (daemonset, deployment) and there they might need it

Reply:
Agree but it's not tested neither.
To avoid pushing dead code, adding single usage of it would be sufficient.

Reply:
```golang
type SecretQuery struct {
	kubeQuery
}

func NewSecretQuery(ctx context.Context, kubeClient client.Client, kubeReader client.Reader, log logr.Logger) SecretQuery {
	return SecretQuery{
		newKubeQuery(ctx, kubeClient, kubeReader, log),
	}
}
```

Reply:
I found something that could be quickly refactored, give me a minute

Reply:
done, kubeReader is now used